newlib/libc/sys/vita/sys/socket.h

```diff
@@ -247,6 +262,24 @@ struct msghdr {
 	int		msg_flags;	/* flags on received message */
 };
 
+#define HAVE_SYS_SOCKET_H
+
+#define HAVE_RECV
```

Comment:
these defines below seem to be curl-specific, why are you adding them to newlib?